bot/src/main/java/edu/java/bot/services/url/strategy/invoker/DomainInvoker.java

```
@@ -0,0 +1,14 @@
+package edu.java.bot.services.url.strategy.invoker;
+
+import edu.java.bot.services.url.strategy.IDomainExecute;
+import jakarta.annotation.Nullable;
+import java.util.Objects;
+import lombok.Setter;
+
+public class DomainInvoker {
+    @Nullable @Setter private IDomainExecute domainCommand;
+
+    public void executeCommand() {
+        Objects.requireNonNull(domainCommand).execute();
```

чтобы не заниматься постоянными проверками, имеет смысле сделать этот класс immutable. То есть, поле domainCommand сделать final, убрать setter, а проверку на null перенести в конструктор. В итоге, в executeCommand ничего проверять не придется, а ошибка будет валиться прямо при создании инстанса. Так сказать fail fast )

---

(3) Решил удалить, пропала необходимость